modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java

```diff
@@ -199,7 +201,13 @@ public URL findResource(String name) {
             // We don't want class loaders in limbo
             synchronized (descriptorMappingLock) {
               if (descriptorMapping.get(matchDescriptor) == null) {
-                URLClassLoader urlClassLoader = new URLClassLoader(new URL[] {url});
+                URLClassLoader urlClassLoader;
+                if (isJavaVersionAtMost(JAVA_1_8)) {
+                  urlClassLoader =
+                      new URLClassLoader(new URL[] {url}, getSystemClassLoader(), new NonCachingURLStreamHandlerFactory());
+                } else {
```

Comment:
this is one way, if we dont like this can we do

` return new URLClassLoader(new URL[] {dependency.getBundleUri().toURL()}, getSystemClassLoader(), getCachingURLStreamHandlerFactory()); `